apps/advisor/pkg/app/utils_test.go

```diff
@@ -117,18 +120,45 @@ func TestProcessCheck_RunError(t *testing.T) {
 	}
 	meta.SetCreatedBy("user:1")
 	client := &mockClient{}
+	typesClient := &mockTypesClient{}
 	ctx := context.TODO()
 
 	check := &mockCheck{
 		items: []any{"item"},
 		err:   errors.New("run error"),
 	}
 
-	err = processCheck(ctx, client, obj, check)
+	err = processCheck(ctx, client, typesClient, obj, check)
 	assert.Error(t, err)
 	assert.Equal(t, "error", obj.GetAnnotations()[checks.StatusAnnotation])
 }
 
+func TestProcessCheck_IgnoreSteps(t *testing.T) {
+	checkType := &advisorv0alpha1.CheckType{}
+	checkType.SetAnnotations(map[string]string{checks.IgnoreStepsAnnotation: "mock"})
+	typesClient := &mockTypesClient{
+		res: checkType,
+	}
+	ctx := context.TODO()
+	check := &mockCheck{
+		items: []any{"item"},
+		err:   errors.New("run error, should not be triggered"),
+	}
+	obj := &advisorv0alpha1.Check{}
+	obj.SetAnnotations(map[string]string{})
+	meta, err := utils.MetaAccessor(obj)
+	if err != nil {
+		t.Fatal(err)
+	}
+	meta.SetCreatedBy("user:1")
+	client := &mockClient{}
+
+	err = processCheck(ctx, client, typesClient, obj, check)
+	assert.NoError(t, err)
+	assert.Equal(t, "processed", obj.GetAnnotations()[checks.StatusAnnotation])
```

Comment:
Nit: I think it would be good to use the consts when possible (IE `StatusAnnotationProcessed`)